clients/js/src/embeddings/VoyageAIEmbeddingFunction.ts

```diff
@@ -0,0 +1,87 @@
+import { IEmbeddingFunction } from "./IEmbeddingFunction";
+
+export enum InputType {
+  DOCUMENT = "document",
+  QUERY = "query"
+}
+
+export class VoyageAIEmbeddingFunction implements IEmbeddingFunction {
+  private model_name: string;
+  private api_url: string;
+  private batch_size: number;
+  private truncation?: boolean;
+  private input_type?: InputType;
+  private headers: { [key: string]: string };
+
+  constructor({
+    voyageai_api_key,
+    model_name,
+    batch_size,
+    truncation,
+    input_type,
+  }: {
+    voyageai_api_key: string;
+    model_name: string;
+    batch_size?: number;
+    truncation?: boolean;
+    input_type?: InputType;
+  }) {
+    this.api_url = "https://api.voyageai.com/v1/embeddings";
+    this.headers = {
+      Authorization: `Bearer ${voyageai_api_key}`,
+      "Content-Type": "application/json",
+    };
+
+    this.model_name = model_name;
+    this.truncation = truncation;
+    this.input_type = input_type;
+    if (batch_size) {
+      this.batch_size = batch_size;
+    } else {
+      if (model_name in ["voyage-2", "voyage-02"]) {
+        this.batch_size = 72;
+      } else {
+        this.batch_size = 7;
+      }
+    }
+  }
+
+  public async generate(texts: string[]) {
+    try {
+      const result: number[][] = [];
+      let index = 0;
+    
+      while (index < texts.length) {
```

Comment:
Well, i think that the Python/TS user doesn't have to know anything about the API itself. What i mean is, we are building the voyageai packages to provide a simpler way to reach our API. This means that i'd do the looping in the voyageai package (and also, i'd create a TS library - it is under development btw). Yes, doing the looping also would mean doing some error handling, where it is possible.
Our python/TS packages are not final yet and atm do not contain the looping nor the error handling, that's why i do the looping here - just like we did elsewhere.

Reply:
Hey @fzowl, my point is that until the looping and error handling are done in a meaningful way, I prefer to keep it simple and useful—no looping. I prefer a failed attempt to embed 10k texts because the max batch size is reached than a partially embedded (and potentially lost) 5k batch due to the rate limit or any other network connection.

The problem is not in your code but rather in the lack of clearer guidance about how to handle these things in Chroma EF wrappers.

When it comes to potential monetary impact on the end user I prefer that we inconvenience them to write a little more error handling/batching code than making the assumption that a simple loop will be enough.

Reply:
Yeah, i i fully understand. I just removed the loop, can you please take a look?